Default computed value for `text-emphasis-style: filled|open` should be `filled|open circle` not `filled|open dot` in horizontal typographic modes

colelao opened the pull request:
#### 1f517ebf0085408001dbf69a61c10c8bb50e3ae6
<pre>
Default computed value for `text-emphasis-style: filled|open` should be `filled|open circle` not `filled|open dot` in horizontal typographic modes
<a href="https://bugs.webkit.org/show_bug.cgi?id=295641">https://bugs.webkit.org/show_bug.cgi?id=295641</a>
<a href="https://rdar.apple.com/155911820">rdar://155911820</a>

Reviewed by Sam Weinig and Tim Nguyen.

Changed default text emphasis style mark to circle when filled or open is specified in horizontal writing mode. Added test cases.

* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/background-clip/clip-text-text-emphasis.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-vertical-lr-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-vertical-lr.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed.html:
* Source/WebCore/style/values/text-decoration/StyleTextEmphasisStyle.cpp:
(WebCore::Style::defaultTextEmphasisMark):

Canonical link: <a href="https://commits.webkit.org/297616@main">https://commits.webkit.org/297616@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ead1c6ca75dcdda8d977a9371fdc9c4907665ffa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112291 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22500 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118369 "Built successfully") | [💥 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62644 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114253 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32675 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40586 "Build is in progress. Recent messages:Running configuration; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85305 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/62644 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115238 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26068 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65735 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25373 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19165 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62215 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95453 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19244 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121695 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39365 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29294 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94114 "Passed tests") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/111786 "Build is in progress. Recent messages:") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39746 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97271 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93938 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24022 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39175 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16973 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35396 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39253 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44741 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38888 "Built successfully") | | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42225 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Running checkout-pull-request; Compiled WebKit (warnings)") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40631 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig reviewed (approved):
Nice. Thanks for picking this up.

colelao committed:
Default computed value for `text-emphasis-style: filled|open` should be `filled|open circle` not `filled|open dot` in horizontal typographic modes
https://bugs.webkit.org/show_bug.cgi?id=295641
rdar://155911820

Reviewed by Sam Weinig and Tim Nguyen.

Changed default text emphasis style mark to circle when filled or open is specified in horizontal writing mode. Added test cases.

* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/background-clip/clip-text-text-emphasis.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-vertical-lr-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed-vertical-lr.html: Copied from LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/parsing/text-emphasis-style-computed.html:
* Source/WebCore/style/values/text-decoration/StyleTextEmphasisStyle.cpp:
(WebCore::Style::defaultTextEmphasisMark):

Canonical link: https://commits.webkit.org/297616@main
